DESIGN_CONSIDERATIONS.md

```diff
@@ -0,0 +1,77 @@
+## Optimize APIs and patterns for developers over designers
+
+Why?
+- There are thousands of them.  The cost of features and bugs is far greater.
```

Comment:
could we, please, be a bit more specific here? What stays behind `them`? the amount of APIs? In that case, could we point to specific examples (briefly, no code is needed) - because, otherwise, it is not absolutely clear in terms of problem's criterias (i.e. 'thousands' - really thousands? what, exactly, makes the situation confusing to the client?)

Reply:
This changes were introduced by @levithomason Will revert them from this PR, and we can create another dedicated PR where he can respond to your questions/comments.